watertap/unit_models/zero_order/tests/test_electroN_P_zo.py

```diff
@@ -0,0 +1,280 @@
+    def test_degrees_of_freedom(self, model):
+        assert degrees_of_freedom(model.fs.unit) == 0
+
+    @pytest.mark.component
+    def test_unit_consistency(self, model):
+        assert_units_consistent(model.fs.unit)
+
+    @pytest.mark.component
+    def test_initialize(self, model):
+        initialization_tester(model)
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_solve(self, model):
+        results = solver.solve(model)
+
+        # Check for optimal solution
+        assert_optimal_termination(results)
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_solution(self, model):
+        assert pytest.approx(1.004, rel=1e-5) == value(
+            model.fs.unit.properties_in[0].flow_vol
+        )
+        assert pytest.approx(0.99602, rel=1e-5) == value(
+            model.fs.unit.properties_in[0].conc_mass_comp["nitrogen"]
+        )
+        assert pytest.approx(0.99602, rel=1e-5) == value(
+            model.fs.unit.properties_in[0].conc_mass_comp["phosphorus"]
+        )
+        assert pytest.approx(0.99602, rel=1e-5) == value(
+            model.fs.unit.properties_in[0].conc_mass_comp["struvite"]
+        )
+
+        assert pytest.approx(0.99636, rel=1e-2) == value(
+            model.fs.unit.properties_treated[0].flow_vol
+        )
+        assert pytest.approx(0.17062, rel=1e-5) == value(
+            model.fs.unit.properties_treated[0].conc_mass_comp["nitrogen"]
+        )
+        assert pytest.approx(0.17062, rel=1e-5) == value(
+            model.fs.unit.properties_treated[0].conc_mass_comp["phosphorus"]
+        )
+
+        assert pytest.approx(0.00183, rel=1e-2) == value(
+            model.fs.unit.properties_byproduct[0].flow_vol
+        )
+        assert pytest.approx(5.4645e-08, rel=1e-5) == value(
+            model.fs.unit.properties_byproduct[0].conc_mass_comp["nitrogen"]
+        )
+        assert pytest.approx(5.4645e-08, rel=1e-5) == value(
+            model.fs.unit.properties_byproduct[0].conc_mass_comp["phosphorus"]
+        )
+        assert pytest.approx(1350.54, abs=1e-5) == value(model.fs.unit.electricity[0])
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_conservation(self, model):
+        for j in model.fs.params.component_list:
+            assert 1e-6 >= abs(
+                value(
+                    model.fs.unit.inlet.flow_mass_comp[0, j]
+                    + sum(
+                        model.fs.unit.generation_rxn_comp[0, r, j]
+                        for r in model.fs.unit.reaction_set
+                    )
+                    - model.fs.unit.treated.flow_mass_comp[0, j]
+                    - model.fs.unit.byproduct.flow_mass_comp[0, j]
+                )
+            )
+
+    @pytest.mark.component
```

Comment:
```suggestion
    @pytest.mark.requires_idaes_solver
    @pytest.mark.component
```

For the macOS tests. That said, I'm not really a fan of testing these string literals.